### PR TITLE
Add AGI v3.2.1

### PR DESCRIPTION
**Pull request by zippozeng**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

**Comment by miccal** [comment]
Thank you for your contribution to `homebrew-cask` @zippozeng.

Could you please try to add a [`zap` stanza](https://docs.brew.sh/Cask-Cookbook#stanza-zap), thanks.

**Comment by zippozeng** [comment]
> Thank you for your contribution to `homebrew-cask` @zippozeng.
> 
> Could you please try to add a [`zap` stanza](https://docs.brew.sh/Cask-Cookbook#stanza-zap), thanks.

added zap trash, please review

File: Casks/agi.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  auto_updates true
+
+  app "AGI.app"
+
+  zap trash: [
+    "~/.agic",
+    "~/Library/Application Scripts/com.apple.siri.parsec.HashtagImagesApp.HashtagImagesExtension",
+    "~/Library/Containers/com.apple.siri.parsec.HashtagImagesApp.HashtagImagesExtension",
+    "~/Library/Staging",
```
Comment (miccal) [inline Casks/agi.rb:24]: Are you certain about these? I have serious doubts that these are related to this Cask.

Comment (zippozeng) [reply Casks/agi.rb:24]: I use nrlquaker/createzap to generate,Is this tap right?

Comment (zippozeng) [reply Casks/agi.rb:24]: how to create zap stanza?

Comment (bevanjkay) [reply Casks/agi.rb:24]: @zippozeng The tool just uses a text search to find matching options, so those lines include `agi`.

It is likely that none of those are relevant. I would recommend using [ `appcleaner](https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/appcleaner.rb)` to find the files that should be removed.

Comment (zippozeng) [reply Casks/agi.rb:24]: I use appcleaner to only find ~/Library/Preferences/net.java.openjdk.cmd.plist and ~/Library/Saved Application State/net.java.openjdk.cmd.saveState, but these two don't seem to be used only by agi

Comment (zippozeng) [reply Casks/agi.rb:24]: I use appcleaner to only find ~/Library/Preferences/net.java.openjdk.cmd.plist and ~/Library/Saved Application State/net.java.openjdk.cmd.saveState, but these two don't seem to be used only by agi

**Comment by miccal** [comment]
Thank you @zippozeng.

**Review by miccal — APPROVED** [review]

**Pull request merged**